src/App.jsx

```diff
@@ -82,6 +82,11 @@ function App() {
   const isSmallScreen = useMediaQuery("(max-width: 600px)");
 
   const { provider, chainID } = useWeb3Context();
+
+  provider.on("network", (_newNetwork, oldNetwork) => {
```

Comment:
Oh i forgot to comment on this. We setup provider listeners on `_initProviders()` under `web3Context.tsx`